coralproject/talk
# [CORL-236] Allow user to delete their account

## What does this PR do?

Adds options to the My Profile > Settings area in the comment stream to allow a user to schedule deletion of their account.

## How do I test this PR?

- Create/use an account you're cool with being deleted
- Head to comment stream
- Select My Profile > Settings > Request account deletion
- Follow modal steps
- See that account is scheduled for deletion
- Test around for state while in "scheduled for deletion" state

### Forcing a deletion

- Update line 19 in `src/core/server/cron/accountDeletion.ts` to `"0,30 * * * * *"` so that deletions are scheduled every 30 seconds
- Update account using Azure: Cosmos DB (or direct mongo queries) so that scheduled deletion is prior to now (should have already happened)

```
"scheduledDeletionDate": {
    "$date": "2019-05-20T20:40:31.627Z" <-- set months ago to account for any UTC day diffs
  },
```

- Run the server `npm run watch`
- Watch as at some point it says it's deleting the account you scheduled for deletion
- Test around to see that the comments are deleted in both stream and moderation area
- Test around the moderation community area to see that the user account is deleted and user drawer is unavailable to be opened
